src/common/libraries/MessageLib.sol

```diff
@@ -130,6 +132,8 @@ library MessageLib {
             length += 2 + message.toUint16(length); //payloadLength
         } else if (kind == uint8(MessageType.UpdateContract)) {
             length += 2 + message.toUint16(length); //payloadLength
+        } else if (kind == uint8(MessageType.UpdateHubContract)) {
```

Comment:
We need special treatment of this inside `messageSourceCentrifugeId` because the poolId in this message does not correlate with the original chain. I think we need to return 0 here.

Reply:
Fixed here: https://github.com/centrifuge/protocol/pull/665